File: backend/src/main/java/com/vocabtai/VocabTai/model/Todo.java
package com.vocabtai.VocabTai.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.ZonedDateTime;

@Entity
@Table(name = "todos")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Todo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String title;
    private String description;

    @Builder.Default
    private Boolean completed = false;

    @Column(name = "user_id")
    private Integer userId;

    @Column(name = "start_time")
    private ZonedDateTime startTime;

    @Column(name = "end_time")
    private ZonedDateTime endTime;

    @Column(name = "created_at")
    private ZonedDateTime createdAt;

    @Column(name = "updated_at")
    private ZonedDateTime updatedAt;
}

File: todolist/src/types/todo.ts
export interface Todo {
  id: number;
  title: string;
  description: string;
  completed: boolean;
  start_time: string;      // ISO string (nên để string, xử lý với dayjs/new Date)
  end_time: string | null; // có thể null
  user_id: number;
  created_at: string;
  updated_at: string;
}


export interface CreateTodoInput {
  title: string;
  description: string;
  user_id: number;
  start_time: string; // ISO format, e.g. "2025-07-05T09:00:00.000Z"
}